Containerfile

```diff
@@ -0,0 +1,32 @@
+# Use the NODE_VERSION argument to specify the Node.js version
+ARG NODE_VERSION
+
+# Use the specified Node.js version as the base image
+FROM node:${NODE_VERSION}-alpine
+
+# Set Node args
+ENV NODE_OPTIONS=--openssl-legacy-provider
+
+RUN apk add --update autoconf automake build-base libtool nasm pkgconf && \
+    rm -rf /var/cache/apk/*
+
+# Setup work directory
+WORKDIR /gitops-website
+
+# Set permissions for non-root user
+RUN chown -R node:node /gitops-website
+
+# Switch user to non-root
+USER node
+
+# Add repo files with correct user:group ownership
+ADD --chown=node:node . .

```

Comment:
We need a `.dockerignore` file with `node_modules` in it. Otherwise we copy the local `node_modules` directory into the container which can result in flaky `npm ci` since we also then have possible development dependencies inside of the container.
@samueltauil is there also a "new" or more general name for `.dockerignore` like `.containerignore` that all the builder tools understand?

Reply:
I'm wondering whether to use the `.dockerignore` name, which will be useful if a potential container user tries to build locally with docker/podman. And as @samueltauil said, the engine also checks this file name.

Reply:
I would be fine running with `.dockerignore` for now.